src/helpers/serializer.ts

```diff
@@ -0,0 +1,91 @@
+import * as firebase from 'firebase/app'
+import {
+  CollectionQueryType,
+  WhereArray,
+  WhereItem,
+  WhereType,
+} from '../types/Query'
+
+export class Serializer {
+  // Helper for where clauses
+  public static multipleConditions<Doc extends object = {}>(
+    w: WhereType<Doc>
+  ): w is WhereArray<Doc> {
+    return !!(w as WhereArray) && Array.isArray(w[0])
+  }
+
+  // Serializer function for where condition
+  private static serializeWhere<Doc extends object = {}>(
+    where: WhereType<Doc>
+  ): WhereType<Doc> {
+    if (this.multipleConditions(where)) {
+      return where.map(w => this.serializeWhere(w)) as WhereArray<Doc>
+    }
+    // Date: Inject serializer options if not specified
+    if (where[2] instanceof Date && !where[3]) {
+      return [...where.slice(0, 3), { type: 'date' }] as WhereItem<Doc>
+    }
+
+    if (where[2] instanceof firebase.firestore.DocumentReference && !where[3]) {
+      return [
+        ...where.slice(0, 2),
+        where[2].path,
+        { type: 'ref' },
+      ] as WhereItem<Doc>
+    }
+
+    return where
+  }
+
+  // Serializer function for query
+  public static serializeQuery<Data extends object = {}>(
+    query: CollectionQueryType<Data>
+  ): string {
+    const { where, ...rest } = query
+
+    return JSON.stringify({
+      where: where ? this.serializeWhere(where) : undefined,
+      ...rest,
+    })
+  }
+
+  // Deserializer function for where condition
+  private static deserializeWhere<Doc extends object = {}>(
+    where: WhereType<Doc>
+  ): WhereType<Doc> {
+    if (this.multipleConditions(where)) {
+      return where.map(w => this.deserializeWhere(w)) as WhereArray<Doc>
+    }
+
+    if (where[3]?.type === 'date' && typeof where[2] === 'string') {
+      return [...where.slice(0, 2), new Date(where[2]), where[3]] as WhereItem<
+        Doc
+      >
+    }
+
+    if (where[3]?.type === 'ref' && typeof where[2] === 'string') {
+      return [
+        ...where.slice(0, 2),
+        firebase.firestore().doc(where[2]),
```

Comment:
I could not find any way of creating a `DocumentReference` without a reference to the firestore instance.
An alternative is passing `fuego` as a parameter to the function and using that instead.
Is that a better solution?